sdcm/nemesis.py

```diff
@@ -2774,7 +2774,15 @@ def choose_snapshot(snapshots_dict):
 
             self.use_nemesis_seed()
             chosen_snapshot_size = random.choice(fitting_snapshot_sizes)
-            snapshot_tag = random.choice(list(snapshot_groups_by_size[chosen_snapshot_size]["snapshots"].keys()))
+            if self.cluster.nodes[0].is_enterprise:
+                snapshot_tag = random.choice(list(snapshot_groups_by_size[chosen_snapshot_size]["snapshots"].keys()))
+            else:
+                oss_regex = r'\d\.\d\.\d'
```

Comment:
This seems fragile as a way to detect OSS versions

In any case, can't we have a double digit patch version?

Reply:
Also the version in the master branch would have `.dev` in it.
I'm not sure if this regex will match it.

Reply:
first of all, this is related to the already backed up snapshot versions, it has nothing to do with system under tests..
what does test if this is enterprise or OSS is `self.cluster.nodes[0].is_enterprise`

anyway, i thought of a better way to represent it, and will do the changes, and push it soon

Reply:
We have better functions to handle this kind of logic, and also we have our own implementation of versions comparing exact for matching those

Anyhow in memory is remove, and 2023.1 doesn't have it's so it should work if we have snapshot from 2023.1

Reply:
> We have better functions to handle this kind of logic, and also we have our own implementation of versions comparing exact for matching those
> 
> Anyhow in memory is remove, and 2023.1 doesn't have it's so it should work if we have snapshot from 2023.1

[these are the snapshots](https://github.com/scylladb/scylla-cluster-tests/blob/master/defaults/manager_persistent_snapshots.yaml) we have... it means that, snapshots from `2022.1.X` and `2022.2.Y` are not usable, not even on enterprise master branch?
if that is the case, a much bigger refactor must be done in here

Reply:
@bhalevy ^^

Reply:
> Anyhow in memory is remove, and 2023.1 doesn't have it's so it should work if we have snapshot from 2023.1

@fruch it wasn't be removed from the schema tables, otherwise upgrade from 2022.X to 2023.1 would not work.

Reply:
> > Anyhow in memory is remove, and 2023.1 doesn't have it's so it should work if we have snapshot from 2023.1
> 
> @fruch it wasn't be removed from the schema tables, otherwise upgrade from 2022.X to 2023.1 would not work.

So it's gonna stay there for good ?

Reply:
We are not actively working on removing it.
I think the best way would be to have the manager restore the schema using some high-level logic.
There are other schema-related incompatibilities between enterprise and open source like IncrementalCompactionStrategy, encryption at rest, and more, that need to be dealt with.

Reply:
changed the logic, running it with latest code [in here](https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/manager/job/restore-nemesis/4/)